src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -851,3 +851,114 @@ def check_read_nwb(self, nwbfile_path: str):
+
+            optical_channels = nwbfile.imaging_planes[self.imaging_plane_name].optical_channel
+            optical_channel_names = [channel.name for channel in optical_channels]
+            assert self.interface_kwargs["channel_name"] in optical_channel_names
+            assert len(optical_channels) == 1
+
+
+class ScanImageSinglePlaneMultiFileImagingInterfaceMixin(DataInterfaceTestMixin, TemporalAlignmentMixin):
+    channel_name = "Channel 1"
+    photon_series_name = "TwoPhotonSeriesChannel1"
+    imaging_plane_name = "ImagingPlaneChannel1"
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.interface_kwargs["channel_name"] = self.channel_name
+        super().__init__(*args, **kwargs)
+
+    def check_read_nwb(self, nwbfile_path: str):
+        with NWBHDF5IO(nwbfile_path, "r") as io:
+            nwbfile = io.read()
+
+            assert self.imaging_plane_name in nwbfile.imaging_planes
+            assert self.photon_series_name in nwbfile.acquisition
+            two_photon_series = nwbfile.acquisition[self.photon_series_name]
+            assert two_photon_series.data.shape == (30, 512, 512)
+            assert two_photon_series.unit == "n.a."
+            assert two_photon_series.data.dtype == np.int16
+            assert two_photon_series.rate is None
+            assert two_photon_series.starting_time is None
+
+            imaging_extractor = self.interface.imaging_extractor
+            times_from_extractor = imaging_extractor._times
+            assert_array_equal(two_photon_series.timestamps[:], times_from_extractor)
+
+            data_from_extractor = imaging_extractor.get_video()
+            assert_array_equal(two_photon_series.data[:], data_from_extractor.transpose(0, 2, 1))
+
+            assert two_photon_series.description == json.dumps(imaging_extractor._imaging_extractors[0].metadata)
+
+            optical_channels = nwbfile.imaging_planes[self.imaging_plane_name].optical_channel
+            optical_channel_names = [channel.name for channel in optical_channels]
+            assert self.interface_kwargs["channel_name"] in optical_channel_names
+            assert len(optical_channels) == 1
+
+
+class ScanImageMultiPlaneImagingInterfaceMixin(DataInterfaceTestMixin, TemporalAlignmentMixin):
+    channel_name = "Channel 1"
+    photon_series_name = "TwoPhotonSeriesChannel1"
+    imaging_plane_name = "ImagingPlaneChannel1"
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.interface_kwargs["channel_name"] = self.channel_name
+        super().__init__(*args, **kwargs)
+
+    def check_read_nwb(self, nwbfile_path: str):
+        with NWBHDF5IO(nwbfile_path, "r") as io:
+            nwbfile = io.read()
+
+            assert self.imaging_plane_name in nwbfile.imaging_planes
+            assert self.photon_series_name in nwbfile.acquisition
+            two_photon_series = nwbfile.acquisition[self.photon_series_name]
+            assert two_photon_series.data.shape == (6, 256, 528, 2)
```

Comment:
Aren't these particular values specific to a single session in the example data? Shouldn't the "expected" values then be set in the main test suite?

If so, then this mixin could have attributes or a dict attribute for each expected value that might change, like `self.expected_two_photon_series_data_shape`, etc.

Reply:
Ah yes, you're right.

Reply:
Okay, I think I managed to simplify the mixin classes with this.